src/main/extras/app/entrypoint.sh

```diff
@@ -162,6 +162,6 @@ export SSL_TRUSTSTORE_DIR
 set -x
 exec java \
     "${FLAGS[@]}" \
-    -cp "$CLASSPATH" \
-    io.cryostat.Cryostat \
+    -cp @/app/jib-classpath-file \
```

Comment:
I think we want the Jib classpath file included *and* at least part of the `$CLASSPATH` env var, since the `$CLASSPATH` currently includes the `$CRYOSTAT_CLIENTLIB_PATH`. Just replacing `$CLASSPATH` with the file seems like it would exclude the clientlib directory from the classpath used in the end. Can this be included in the pom.xml config for Jib, maybe, so that the file content also includes the clientlib path? If not then I think joining the clientlib path with the Jib file together would work.

Reply:
How can I check that my last commit contains the clientlib path in `/app/jib-classpath-file`?

Reply:
I like to use dive: https://github.com/wagoodman/dive

ex. `dive podman://quay.io/cryostat/cryostat:latest`

This lets you see the contents within an image by listing directories and filenames/permissions/etc, although I'm not sure it will show you the contents of specific files. You could use `podman export` for that if not.

Reply:
Otherwise, a cheap hack would be to just modify the `entrypoint.sh` to do something like `cat /app/jib-classpath-file` so that the file contents get dumped to stdout when you run the container.

Reply:
There's an open [feature request](https://github.com/wagoodman/dive/issues/336) to view files in Dive.

In the meantime, `podman export` shows that `${env.CLASSPATH}:/clientlib` gets placed in front of Jib's computed classpath: [github gist link](https://gist.githubusercontent.com/jan-law/336111fcf6980397054bb997a8dc548c/raw/fc8c4cfef88208a2b7122dd916c0aad8ffec012a/jib-classpath-file)

Reply:
`${env.CLASSPATH}:/clientlib`

That looks very wrong to me - I would have thought that it would at least be expanded out, not inserted as a literal string into the classpath. Does `java -cp` even accept that? It's either going to cause the command to choke and fail to start, or it will be ignored and may as well be removed.

Reply:
Is it possible to do `java -cp @/app/jib-classpath-file:/something:/another/directory`? ie, to use the `@` syntax to read from a file and also to append to it using the typical semicolon separator to directly add additional paths?

`entrypoint.sh` currently does this:

```
CLASSPATH="/app/resources:/app/classes:/app/libs/*"
if [ -n "CRYOSTAT_CLIENTLIB_PATH" ]; then
    CLASSPATH="$CLASSPATH:$CRYOSTAT_CLIENTLIB_PATH/*"
fi
```

so I wonder if this can just be replaced by:

```
CLASSPATH="@/app/jib-classpath-file"
if [ -n "CRYOSTAT_CLIENTLIB_PATH" ]; then
    CLASSPATH="$CLASSPATH:$CRYOSTAT_CLIENTLIB_PATH/*"
fi
```

this allows the `$CRYOSTAT_CLIENTLIB_PATH` to still be set at runtime, rather than requiring it to be hardcoded to `/clientlib`.

Reply:
`entrypoint.sh`:
```
CLASSPATH="@/app/jib-classpath-file"
if [ -n "CRYOSTAT_CLIENTLIB_PATH" ]; then
    CLASSPATH="$CLASSPATH:$CRYOSTAT_CLIENTLIB_PATH/*"
fi
...

exec java \
    "${FLAGS[@]}" \
    -cp "$CLASSPATH" \
    @/app/jib-main-class-file \
    "$@"
```
Output: `Error: could not open `/app/jib-classpath-file:/clientlib/*'`

I also tried
`echo "$CRYOSTAT_CLIENTLIB_PATH/*" >> @/app/jib-classpath-file`
which gives me
`/app/entrypoint.sh: line 158: @/app/jib-classpath-file: No such file or directory`

Reply:
That `echo` line should work, but you would have to remove the `@`. That's special syntax that the `java` command uses to know that the argument passed to the `-cp` flag is a file containing text defining the classpath, rather than the argument simply being the text of the classpath itself.

I think you would also need to prepend the `:` separator, so that the clientlib path defined by the env var is correctly appended/separated from the pre-existing file contents.

Reply:
> if [ -n "$CRYOSTAT_CLIENTLIB_PATH" ]; then
>   echo -n ":$CRYOSTAT_CLIENTLIB_PATH/*" >> /app/jib-classpath-file
> fi
> 
> exec java -cp @/app/jib-classpath-file

Hmm, I don't have permissions to edit the file:
`/app/entrypoint.sh: line 155: /app/jib-classpath-file: Permission denied`

I'll also take a look at Jib's github issues and let you know if I find anything.

Reply:
Ah. That's unfortunate. You could try this then:

```
CLASSPATH="$( cat /app/jib-classpath-file )"
if [ -n "CRYOSTAT_CLIENTLIB_PATH" ]; then
    CLASSPATH="$CLASSPATH:$CRYOSTAT_CLIENTLIB_PATH/*"
fi

exec java -cp "$CLASSPATH"
```

Presumably the uid has access to at least read the jib-classpath-file, so this should work around lacking write permissions for it.

If you have read permissions you could also read and copy the contents to a different, new file. Then conditionally append the additional clientlib path to that new file, and then pass the path to the new file with the `java -cp @newfile`.

Reply:
>CLASSPATH="$( cat /app/jib-classpath-file )"

This approach works:
```
+ exec java ...(flags)... -cp '/app/resources:/app/classes:/app/libs/cryostat-core-2.3.0.jar:
...(more jars)...
:/app/libs/jsoup-1.13.1.jar:/clientlib/*' @/app/jib-main-class-file
```

I don't have copy permissions to create a new file, so I'll stick to the above solution.
`cp: cannot create regular file '/app/extended-classpath-file': Permission denied`

Reply:
The uid running inside the container probably just doesn't have write permissions inside `/app`, but you could probably write inside of `/tmp` instead, or somewhere else. Anyway, just reading the generated contents into an env var seems fine. It seems pretty unlikely that the classpath length would exceed the ~32k character limit of an env var. If it ever does (!) we'll find a way to do it by file copy/append instead, again.